src/FortuneWindow.cpp

```diff
@@ -12,76 +13,66 @@
-	back->SetViewColor(ui_color(B_PANEL_BACKGROUND_COLOR));
-	AddChild(back);
+	BButton* next = new BButton("nextbutton","Get Another",
+		new BMessage(M_GET_ANOTHER_FORTUNE));
 
-	BButton *close = new BButton(BRect(0,0,1,1),"closebutton","Close",
-								new BMessage(B_QUIT_REQUESTED), B_FOLLOW_RIGHT | B_FOLLOW_BOTTOM);
-	close->ResizeToPreferred();
-	close->MoveTo(Bounds().right - 15 - close->Frame().Width(), 
-				Bounds().bottom - 15 - close->Frame().Height());
-	back->AddChild(close);
-	close->MakeDefault(true);
-	
-	BButton *next = new BButton(BRect(0,0,1,1),"nextbutton","Get Another",
-								new BMessage(M_GET_ANOTHER_FORTUNE), B_FOLLOW_RIGHT | B_FOLLOW_BOTTOM);
-	next->ResizeToPreferred();
-	next->MoveTo(close->Frame().left - 15 - next->Frame().Width(), 
-				Bounds().bottom - 15 - next->Frame().Height());
-	back->AddChild(next);
-	
-	BButton *about = new BButton(BRect(0,0,1,1),"aboutbutton","About" B_UTF8_ELLIPSIS,
-								new BMessage(M_ABOUT_REQUESTED), B_FOLLOW_LEFT | B_FOLLOW_BOTTOM);
-	about->ResizeToPreferred();
-	about->MoveTo(next->Frame().left - 15 - about->Frame().Width(), 
-				Bounds().bottom - 15 - about->Frame().Height());
-	back->AddChild(about);
-	
-	
-	BRect r(15,15,Bounds().right - B_V_SCROLL_BAR_WIDTH - 15, next->Frame().top - 15);
-	fTextView = new BTextView(r,"textview",r.OffsetToCopy(0,0).InsetByCopy(10,10),B_FOLLOW_ALL,
-								B_WILL_DRAW | B_PULSE_NEEDED | B_FRAME_EVENTS);
+	BButton* about = new BButton("aboutbutton","About" B_UTF8_ELLIPSIS,
+		new BMessage(M_ABOUT_REQUESTED));
+
+	fTextView = new BTextView("textview", B_WILL_DRAW | B_PULSE_NEEDED | B_FRAME_EVENTS);
 	fTextView->MakeEditable(false);
-	
-	BScrollView *sv = new BScrollView("scrollview",fTextView,B_FOLLOW_ALL,0,false,true);
-	back->AddChild(sv);
-	
+	fTextView->SetInsets(10, 10, 10, 10);
+
+	BScrollView *sv = new BScrollView("Fortune",
+		fTextView, 0, false, true);
+
 	BString fortune;
 	status_t status = fFortune.GetFortune(&fortune);
-	if(status == B_OK)
+
+	if (status == B_OK)
 	{
 		BString title;
 		fFortune.LastFilename(&title);
 		title.Prepend("Fortune: ");
 		SetTitle(title.String());
-		
+
 		fTextView->SetText(fortune.String());
 	}
-	else	
+	else
 	{
 		fTextView->SetText("Fortuna had a problem getting a fortune.\n\n"
 			"Please make sure that you have installed fortune files to "
 			"the folder ");
 		fTextView->Insert(path.Path());
 		fTextView->Insert(".");
 	}
-	
+
 	next->MakeFocus(true);
-	
-	// Calculate a good width for the window
-	SetSizeLimits(45 + close->Bounds().Width() + next->Bounds().Width(),30000,200,30000 ) ;
-	r = BScreen().Frame();
-	MoveTo((r.Width()-Bounds().Width())/2,r.Height() / 4);
+	next->MakeDefault(true);
+
+	BLayoutBuilder::Group<>(this, B_VERTICAL, B_USE_DEFAULT_SPACING)
+		.SetInsets(B_USE_WINDOW_INSETS)
+		.Add(sv)
+		.AddGroup(B_HORIZONTAL)
+			.AddGlue()
+			.AddGlue()
+			.Add(next)
+			.Add(about)
+			.AddGlue()
+			.AddGlue()
+			.AddGlue()
```

Comment:
You added more glues instead of removing the extra ones. One and only one glue on either side of the 2-button group.